src/main/java/org/jenkins/plugins/lockableresources/LockStep.java

```diff
@@ -11,25 +11,47 @@
 import hudson.Extension;
 import hudson.model.AutoCompletionCandidates;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+
 public class LockStep extends AbstractStepImpl implements Serializable {
 
-	public final String resource;
+	@CheckForNull
+	public String resource = null;
+
+	@CheckForNull
+	public String label = null;
+
+	public int quantity = 0;
 
 	public boolean inversePrecedence = false;
 
 	@DataBoundConstructor
-	public LockStep(String resource) {
-		if (resource == null || resource.isEmpty()) {
-			throw new IllegalArgumentException("must specify resource");
-		}
-		this.resource = resource;
+	public LockStep() {
```

Comment:
This is backward incompatible, but I fear there is no way to preserve it in this case.
Any call to to `lock` in the form `lock ('my-resource') { ... }` is going to fail.

@jglick any idea on how keep backward compatibility here?